src/repr/src/adt/timestamp.rs

```diff
@@ -859,6 +992,86 @@ mod test {
         assert_eq!(result, Interval::new(-months, 0, 0));
     }
 
+    fn assert_round_to_precision(
+        mut dt: CheckedTimestamp<NaiveDateTime>,
+        precision: u8,
+        expected: i64,
+    ) {
+        dt.round_to_precision(Some(TimestampPrecision(precision)));
+        assert_eq!(expected, dt.timestamp_micros());
+    }
+
+    #[mz_ore::test]
+    fn test_round_to_precision() {
+        let date = CheckedTimestamp::try_from(
+            NaiveDate::from_ymd_opt(1970, 1, 1)
+                .unwrap()
+                .and_hms_nano_opt(0, 0, 0, 123456789)
+                .unwrap(),
+        )
+        .unwrap();
+        assert_round_to_precision(date, 0, 0);
+        assert_round_to_precision(date, 1, 100000);
+        assert_round_to_precision(date, 2, 120000);
+        assert_round_to_precision(date, 3, 123000);
+        assert_round_to_precision(date, 4, 123500);
+        assert_round_to_precision(date, 5, 123460);
+        assert_round_to_precision(date, 6, 123456);
```

Comment:
I just noticed, `123456789` should have been rounded here to `123457`. Correcting this.

Reply:
done!